presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -429,7 +432,7 @@ public ReadContext init(
             ImmutableList.Builder<parquet.schema.Type> fields = ImmutableList.builder();
             for (HiveColumnHandle column : columns) {
                 if (!column.isPartitionKey() && column.getHiveColumnIndex() < messageType.getFieldCount()) {
-                    fields.add(messageType.getType(column.getName()));
+                    fields.add(getParquetType(configuration, column, messageType));
```

Comment:
Can you explain how the reordering works?  I expect to see code that translates from the `HiveColumnHandle.hiveColumnIndex` to the ordinal in the Parquet file (based on name matching).